kong/clustering/services/sync/hooks.lua

```diff
@@ -0,0 +1,191 @@
+local _M = {}
+local _MT = { __index = _M, }
+
+
+local hooks = require("kong.hooks")
+local EMPTY = require("kong.tools.table").EMPTY
+
+
+local ipairs = ipairs
+local ngx_null = ngx.null
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_DEBUG = ngx.DEBUG
+
+
+local DEFAULT_PAGE_SIZE = 512
+
+
+function _M.new(strategy)
+  local self = {
+    strategy = strategy,
+  }
+
+  return setmetatable(self, _MT)
+end
+
+
+local function get_all_nodes_with_sync_cap()
+  local res, err = kong.db.clustering_data_planes:page(DEFAULT_PAGE_SIZE)
+  if err then
+    return nil, "unable to query DB " .. err
+  end
+
+  if not res then
+    return EMPTY
+  end
+
+  local ret = {}
+  local ret_n = 0
+
+  for _, row in ipairs(res) do
+    for _, c in ipairs(row.rpc_capabilities) do
+      if c == "kong.sync.v2" then
+        ret_n = ret_n + 1
+        ret[ret_n] = row.id
```

Comment:
Upon hitting this branch, we know we have found the capability for the row, so we can break out of the `for _, c in ipairs(row.rpc_capabilities) do` loop:

```suggestion
        ret[ret_n] = row.id
        break
```

Reply:
No, the purpose of this loop is finding all nodes which support `kong.sync.v2` rpc.

Reply:
done.